feat(breakingchange): :sparkles: add breaking change configuration

yi-Xu-0100 opened the pull request:
1. add `detectBreakingChange` option.
2. add `promptBreakingChange` option.
3. add `breakingChangeFormat` option.
4. fix the prompt context retaining issue.

close #39, close  #100

vivaxy reviewed (commented):
👍 That's a lot work done here. I'd like to test it for some time.

yi-Xu-0100 commented:
@vivaxy I test many times today and will use it in the future for some time. During the test, I found the bug (#100). I try to delete the export function and use the constructor in the prompt function to fix the bug.

I do not why it can work, but it fixes the issue. Maybe we can make this little commit merge into master and make a patch firstly. See more in (#100) and the commit (326c953). 😀

vivaxy commented on package.nls.json:
```diff
@@ -20,17 +25,27 @@
+  "extension.sources.prompt.breakingChange.placeholder": "Select a way to add breaking changes of this change.",
+  "extension.sources.prompt.breakingChange.noneItem.label": "None",
+  "extension.sources.prompt.breakingChange.noneItem.detail": "No breaking change.",
+  "extension.sources.prompt.breakingChange.pointItem.label": "! (point mode)",
```
Why is this called `point mode`?

vivaxy replied:
Maybe it can be called as `Breaking change in header`.

yi-Xu-0100 replied:
`! exclamation mark (英式英语) / point (美式英语)`

I misunderstood it. The `exclamation point` may be correct. 😅

I will use `Breaking change in header`, `Breaking change in footer` and `Breaking change as token in footer`  for the labels of them.

vivaxy commented:
> @vivaxy I test many times today and will use it in the future for some time. During the test, I found the bug (#100). I try to delete the export function and use the constructor in the prompt function to fix the bug.
> 
> I do not why it can work, but it fixes the issue. Maybe we can make this little commit merge into master and make a patch firstly. See more in (#100) and the commit ([326c953](https://github.com/vivaxy/vscode-conventional-commits/commit/326c953cdf2cf25ca8946347da38e7a627990be8)). 😀

Maybe because `commitMessage` is in singleton, and the extension not exited after committing. We can check the output to make sure. If so, it's a nice fix.

yi-Xu-0100 committed:
feat(breakingchange): :sparkles: add  breaking change section

1. add `detectBreakingChange` option.
2. add `promptBreakingChange` option.
3. add two `breakingChangeFormat`.

silopolis commented:
Any chance of this getting merged someday ?

yi-Xu-0100 commented:
👋 @silopolis @vladly I'm glad you noticed this issue. It hasn't been updated in a long time, and fixing it will require some time. Currently, I don't have the leisure time to address this problem. If you have any ideas, you can submit a pull request to help with the development. 😊

Moreover, due to the long-term use of the extension, frequent commits with disruptive changes are unreasonable. Therefore, this pull request has been put on hold and considered a low-priority feature. /🤔

leonardvgend commented:
Hello, if I understand correctly, this change will autmoatically add "BREAKING CHANGE: <your breaking change here>" when the user enters their breaking change into this prompt?

![image](https://github.com/vivaxy/vscode-conventional-commits/assets/112492621/a351b87a-f875-4426-ba2d-b8338c6116db)

If so, this would be an amazing feature! I actually expected it to work like that already, and I thought something might be wrong with my configs.

sag1v commented:
Hi, is this going to be merged soon?

philipp-winterle commented:
As this is part of Conventional Commits as a schema BREAKING CHANGE is a mandatory feature to be in line with the defaults. I would not consider this as a low prio feature. 

With this extension no major release would be possible if not done manual
